app/models/nic/base.rb

```diff
@@ -210,16 +230,31 @@ def sync_name
       synchronizer.sync_name if synchronizer.sync_required?
     end
 
-    def validate_host_location
+    def validate_host_subnet_location
       return true if self.host.location.nil? && self.subnet.locations.empty?
       errors.add(:subnet, _("is not defined for host's location.")) unless include_or_empty?(self.subnet.locations, self.host.location)
     end
 
-    def validate_host_organization
+    def validate_host_subnet_organization
       return true if self.host.organization.nil? && self.subnet.organizations.empty?
       errors.add(:subnet, _("is not defined for host's organization.")) unless include_or_empty?(self.subnet.organizations, self.host.organization)
     end
 
+    def validate_host_subnet6_location
+      return true if self.host.location.nil? && self.subnet6.locations.empty?
+      errors.add(:subnet6, _("is not defined for host's location.")) unless include_or_empty?(self.subnet6.locations, self.host.location)
+    end
+
+    def validate_host_subnet6_organization
+      return true if self.host.organization.nil? && self.subnet6.organizations.empty?
+      errors.add(:subnet6, _("is not defined for host's organization.")) unless include_or_empty?(self.subnet6.organizations, self.host.organization)
+    end
+
+    def validate_subnets_types
+      errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if self.subnet.present? && self.subnet.type != 'Subnet::Ipv4'
+      errors.add(:subnet6, _("must be of type Subnet::Ipv6.")) if self.subnet6.present? && self.subnet6.type != 'Subnet::Ipv6'
+    end
```

Comment:
Does this trigger when you enter IPv4 address into IPv6 field? I am not sure if I get when we get this error. Anyway, since this is user facing error for sure, I'd perhaps suggest something less technical (e.g. "must be an IPv6 address").